rust/src/client.rs

```diff
@@ -156,12 +156,30 @@ pub(crate) fn client_handle_fd_argument(arg: &str, arch: &str) -> CxxResult<Vec<
 /// Basically we load too much data before claiming the bus name,
 /// and dbus doesn't give us a useful error.  Instead, let's talk
 /// to systemd directly and use its client tools to scrape errors.
+///
+/// What we really should do probably is use native socket activation.
 pub(crate) fn client_start_daemon() -> CxxResult<()> {
     let service = "rpm-ostreed.service";
     // Assume non-root can't use systemd right now.
     if rustix::process::getuid().as_raw() != 0 {
         return Ok(());
     }
+    // Unfortunately, RHEL8 systemd will count "systemctl start"
+    // invocations against the restart limit, so query the status
+    // first.
+    let activeres = Command::new("systemctl")
+        .args(&["is-active", "rpm-ostreed"])
+        .output()?;
+    if !activeres.status.success() {
```

Comment:
`is-active` returns nonzero if the service is not active. But even then, I think it'd be better to not make this a hard error.

So maybe let's drop this if-statement entirely and key off purely on its stdout regardless of the exit code?

Reply:
Argh, I am so used to my custom fish prompt that clearly shows exit status of last command that I get tripped up in bash cases when it isn't present.